libpoolprotocols/getwork/EthGetworkClient.cpp

```diff
@@ -21,9 +21,10 @@ EthGetworkClient::~EthGetworkClient()
 
 void EthGetworkClient::connect()
 {
-	if (m_connection_changed) {
-		p_client = new ::JsonrpcGetwork(new jsonrpc::HttpClient(m_host));
-	}
+	if (m_connection_changed)
+		p_client = new ::JsonrpcGetwork(new jsonrpc::HttpClient(
```

Comment:
@smurfy Where s p_client deleted? Shouldn't we delete it 1st before reallocating here if not nullptr?